packages/firestore/src/platform_browser/webchannel_connection.ts

```diff
@@ -301,10 +301,19 @@ export class WebChannelConnection implements Connection {
         // can be removed once the fix has been rolled out.
         const error = msgData.error || (msgData[0] && msgData[0].error);
         if (error) {
-          log.debug(LOG_TAG, 'WebChannel received error:', error);
           // error.status will be a string like 'OK' or 'NOT_FOUND'.
           const status: string = error.status;
           let code = mapCodeFromRpcStatus(status);
+          if (code === Code.RESOURCE_EXHAUSTED) {
```

Comment:
@mikelehen: or should we just log.error this regardless (excepting Code.OK)?

Reply:
Some errors are expected (e.g. when a transaction fails due to concurrent writes, we get a FAILED_PRECONDITION error or similar) so we don't want to log all errors.

Also, we have multiple network transports (webchannel_connection.ts for the browser, grpc_connection.ts for node).  So it may make sense to tweak the log message here instead: https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/remote/persistent_stream.ts#L321  (that code is used by both our write stream and watch stream, regardless of transport)

Reply:
Done.

Completely unrelated to this change, I just noticed that log.debug() takes a tag parameter, but log.error() does not. Is that on purpose? (A quick grep through the code base suggests we always supply a tag when using log.debug. However, we seem to also supply a tag when using log.error() roughly a third of the time.) Given that log.error() takes a va_list anyway, it'll obviously still work; it just seems a little odd.

Reply:
Hah...  Uh, I'm not 100% sure. My guess is the goal with the tag is to be able to theoretically enable/disable the debug logs on a per-tag basis (this isn't actually possible currently).  But error messages are always-on, much fewer, and are more user-facing, so the tag isn't needed.  We should probably clean up the usages of log.error() that include a tag in that case though.  Perhaps open a P3 / internal cleanup bug (or just make a PR :-)) if you feel like it.

Reply:
https://github.com/firebase/firebase-js-sdk/pull/376